File: LeetCode/1512. Number of Good Pairs/solution.py
"""
Runtime: 36 ms
Memory Usage: 15 MB
"""
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        m = collections.Counter(nums)
        return sum(v * (v - 1) // 2 for k, v in m.items())

File: LeetCode/1512. Number of Good Pairs/solution.cpp
/**
 * Runtime: 4 ms
 * Memory Usage: 6.9 MB 
 */
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int ans = 0;
        for (int i = 0; i < nums.size(); ++ i) {
            for (int j = i + 1; j < nums.size(); ++ j) {
                if (nums[i] == nums[j]) {
                    ++ans;
                }
            }
        }
        return ans;
    }
};
